client/dialogs/dialog-reducer.ts

```diff
@@ -23,6 +23,18 @@ export default immerKeyedReducer(DEFAULT_DIALOG_HISTORY_STATE, {
   },
 
   ['@dialogs/close'](state, action) {
-    state.history.pop()
+    const { dialogType } = action.payload
+
+    if (dialogType === 'all') {
+      state.history = []
+      return
+    }
+
+    const dialogIndex = state.history.findIndex(h => h.type === dialogType)
```

Comment:
`findIndex` finds the first index that matches, but the way you're stacking dialogs is in ascending order of time opened, that is, the oldest dialogs are first. When a dialog of a particular type is closed, we should be closing the *newest* dialog of that type, not the *oldest*. You want `findLastIndex`, which doesn't exist, so you'll need to implement this behavior yourself.